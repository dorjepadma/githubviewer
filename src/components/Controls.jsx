import React from 'react';
import PropTypes from 'prop-types';

const Controls = ({ text, onChange, onSubmit, inputName = 'text', buttonText = 'Submit' }) => (
  <form onSubmit={onSubmit}>
    <input type="text" name={inputName} value={text} onChange={onChange} />
    <button variant="primary">{buttonText}</button>
  </form>
);

Controls.propTypes = {
  text: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  inputName: PropTypes.string,
  buttonText: PropTypes.string
};

export default Controls;
