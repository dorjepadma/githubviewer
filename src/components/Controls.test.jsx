import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls';

describe('Controls component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Controls text="hi" onChange={() => {}} onSubmit={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
