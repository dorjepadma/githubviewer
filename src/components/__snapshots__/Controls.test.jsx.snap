// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders App 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="text"
    onChange={[Function]}
    type="text"
    value="hi"
  />
  <button
    variant="primary"
  >
    Submit
  </button>
</form>
`;
